### PR TITLE
Ye updated ci

### PR DESCRIPTION
**Pull request by yawetse**
- added pre-commit hooks for conventional commits

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,14 +7,14 @@
...
-    "postinstall": "(cd docs && yarn install); (cd scikitjs-browser && yarn install); (cd scikitjs-node && yarn install); (cd shared && yarn install)"
+    "test:clean": "npm run test:node && npm run test:browser",
+    "test:browser": "cd scikitjs-browser && npm run test",
+    "test:node": "cd scikitjs-node && npm run test",
+    "build:browser": "cd scikitjs-browser && npm run build",
+    "build:node": "cd scikitjs-node && npm run build",
+    "build": "npm run build:node && npm run build:browser",
+    "build:docs": "cd docs && npm run build",
+    "postinstall": "(cd docs && npm install); (cd scikitjs-browser && npm install); (cd scikitjs-node && npm install); (cd shared && npm install); (npx husky install);"
   },
```
Comment (dcrescim) [inline package.json:18]: The switch to npm looks great. Love the husky addition as well.

File: package.json [hunk 2]
```diff
@@ -27,12 +27,14 @@
...
     "@rollup/plugin-commonjs": "^21.0.1",
     "@rollup/plugin-node-resolve": "^13.0.6",
     "@rollup/plugin-typescript": "^8.3.0",
     "eslint": "^8.2.0",
+    "husky": "^7.0.4",
     "rollup-plugin-terser": "^7.0.2",
-    "typedoc": "^0.22.9",
-    "yarn": "^1.22.17"
+    "typedoc": "^0.22.9"
   }
```
Comment (dcrescim) [inline package.json:39]: Looks great. Glad you removed yarn from the package.json. We should also switch the package.json's in scikitjs-browser and scikitjs-node right? The more I think about it, we should prob just switch everything over to npm and standardize on that.

**Review by dcrescim — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (yawetse) [reply package.json:39]: Definitely, I wanted to just remove it. Step. By step and make sure the current CI wasn’t breaking too much

**Comment by yawetse** [comment]
Thanks for the nvmrc fix, I want to make sure not too many things are breaking as I update the CI otherwise it’ll be annoying fixing all the broken tests/checks

File: package.json (hunk 2 above)
Comment (dcrescim) [reply package.json:39]: Sounds good. I'm happy with this PR then. Merge whenever you'd like. Great work.

**Comment by dcrescim** [comment]
Totally agree @yawetse. Feel free to merge if you're happy. Looks great to me.

**Pull request merged**